internal/manifests/collector/ports.go

```diff
@@ -158,27 +162,33 @@ func getReceiverServicePort(logger logr.Logger, serviceAddress string, receiverN
 				Port:     receiverDefaultPortsMap[receiverName],
 				Protocol: protocol,
 			}
-			servicePortsMap[receiverDefaultPortsMap[receiverName]] = sp
+			servicePortsMap[receiverDefaultPortsMap[receiverName]] = []corev1.ServicePort{sp}
 		}
 	}
 }
 
-func getLogsReceiversServicePorts(logger logr.Logger, config *adapters.CwaConfig, servicePortsMap map[int32]corev1.ServicePort) {
+func getLogsReceiversServicePorts(logger logr.Logger, config *adapters.CwaConfig, servicePortsMap map[int32][]corev1.ServicePort) {
 	//EMF - https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format_Generation_CloudWatch_Agent.html
 	if config.Logs != nil && config.Logs.LogMetricsCollected != nil && config.Logs.LogMetricsCollected.EMF != nil {
 		if _, ok := servicePortsMap[receiverDefaultPortsMap[EMF]]; ok {
 			logger.Info("Duplicate port has been configured in Agent Config for port", zap.Int32("port", receiverDefaultPortsMap[EMF]))
 		} else {
-			sp := corev1.ServicePort{
-				Name: EMF,
-				Port: receiverDefaultPortsMap[EMF],
+			tcp := corev1.ServicePort{
+				Name:     EMFTcp,
```

Comment:
Does the port Name need to correspond to anything else?  [Docs](https://kubernetes.io/docs/reference/kubernetes-api/service-resources/service-v1/#ServiceSpec) say "When considering the endpoints for a Service, this must match the 'name' field in the EndpointPort."  Does that apply for us?

Reply:
Spoke with @lisguo offline, we do not create Endpoints in the operator, so there's nothing to worry about here.